design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,183 @@
+
+One ConfigMap will be created per ObjectBucketClaim instance.  ConfigMap names will be semi-deterministic and derived from the ObjectBucketClaim name.  If an ObjectBucketClaim is created with the `generateName` field, then the name cannot be known ahead of time.
+
+Additionally, the ObjectBucketClaim and Pod can be deployed at the same time thanks to in-built synchronization in Kubernetes. Pods mounting the configMap as either `envFrom:` or `env[0].valueFrom.ConfigMapKeyRef` will have a status of `CreateContainerConfigError` until the configMap is created. Pods mounting as a volume will have a status of `ContainerCreating` until the configMap is created. In both cases, the Pod will start once the configMap exists.
+
+Bucket deletion in the early stages of this controller design will not be addressed beyond cleaning up API objects.  Deletion of the bucket within the object store is left as an administrator responsibility.  This is to prevent the accidental destruction of data by users.
+
+## Design
+
+#### ConfigMap Names
+
+ConfigMaps generated by the ObjectBucketClaim Operator will be named in the pattern:
+
+    objectBucketClaim.metadata.name-config-map
+
+For instance, a claim with `metadata.name: "my-object-bucket-123"` will be named `my-object-bucket-123-config-map`
+
+### Work Flow
+
+#### Assumptions
+
+- An object store has been provisioned with a reachable endpoint.  ([e.g. the Rook-Ceph Object provisioner](https://rook.github.io/docs/rook/master/ceph-object.html)) This may also be an external endpoint such as AWS S3.
+- A Service and/or Endpoint API object exists to route connections to the object store.  Typically, object store operators generate these. In cases where the is external to the cluster, they must be configured manually.
+
+#### Use Cases
+
+**Use Case: Expose an Object Store Endpoint**
+
+_As a cluster admin, I want to expose an object store to cluster users so that they may begin bucket provisioning via the Kubernetes API._
+
+![admin-actions](./obc-admin.png)
+
+These steps may be performed in any order.
+
+The admin ...
+
+- creates the ObjectBucketClaim Operator (a Deployment API object)
+- creates a StorageClass with `parameters[serviceName]=objectstore-service` and `parameters[serviceNamespace]=some-namespace`
+  - `provisioner` field is left blank.  Since only the ObjectBucketClaim operator is watching ObjectBucketClaim instances, defining the operator is unnecessary. 
+
+Once the operator is running, it will be begin watching ObjectBucketClaims.
+- An object store has been provisioned with a reachable endpoint.  ([e.g. Rook-Ceph Object](https://rook.github.io/docs/rook/master/ceph-object.html))
+
+#### MVP Use Cases
+
+**Use Case: Expose an Object Store Endpoint**
+
+_As a Kubernetes admin, I want to expose an object store to cluster users for bucket provisioning via the Kubernetes API._
```

Comment:
I don't see a difference between this use-case and the one above.

Reply:
editing mistake, will fix